Lib/_aix_support.py

```diff
@@ -0,0 +1,77 @@
+"""Shared AIX support functions."""
+
+import sys
+import subprocess
+from sysconfig import get_config_var
+
+"""
+AIX filesets are identified by four decimal values: V.R.M.F.
+V (version) and R (release) can be retreived using ``uname``
+M(aintenance) and F(ix) values are not available via uname
+For determining ABI compatibility the values V.R.M
+are essential - suplemented by a final value known as
+`builddate` that is expressed as YYWW (Year WeekofYear)
+
+The values for V.R.M of the system Python was built on are available in
+the variable sysconfig.get_config_var("BUILD_GNU_TYPE"). The system builddate
+value may be available in sys.config.get_config_var("AIX_BUILDDATE")
+When all of these values on a running system are greater or equal to the
+values of the build system then the bdist modules are binary compatible
+with the AIX kernel.
+
+For pep425 purposes the AIX platform tag becomes:
+"AIX-{}-{:04d}-{}".format(vmtl, builddate, bitsize)
+e.g., "AIX-6107-1415-32" for AIX 6.1 TL7 bd 1415, 32-bit
+and, "AIX-6107-1415-64" for AIX 6.1 TL7 bd 1415, 64-bit
+"""
+
+# _bd - set to impossible setting year98-week98 if None
+_tmp = str(get_config_var("AIX_BUILDDATE"))
+_bd = 9898 if (_tmp == "None") else int(_tmp)
+_bgt = get_config_var("BUILD_GNU_TYPE")
+_is_32bit = sys.maxsize == 2147483647
+
+
+def _aix_tag(vrtl, bd):
+    # type: (int, int) -> str
+    sz = 32 if _is_32bit else 64
+    return "AIX-{}-{:04d}-{}".format(vrtl, bd, sz)
+
+
+# compute vrtl from the VRMF string
+def _aix_vrtl(vrmf):
+    # type: (str) -> int
+    v, r, tl = vrmf.split(".")[:3]
+    return int("{}{}{:02d}".format(v[-1], r, int(tl)))
+
+
+def _aix_bosmp64():
+    # type: () -> List[str]
+    """
+    The fileset bos.mp64 is the AIX kernel. It's VRMF and builddate
+    reflect the current levels of the runtime environment.
+    """
+    tmp = subprocess.check_output(["/usr/bin/lslpp", "-Lqc", "bos.mp64"])
```

Comment:
If you use `text=True`, it'll return Unicode so you won't have to `.decode()` it manually:
```suggestion
    lslpp_output = subprocess.check_output(["/usr/bin/lslpp", "-Lqc", "bos.mp64"], text=True)
```
Also, `lslpp_output` is more verbose about what the result variable holds.

Reply:
commented below - earlier code was over 80 characters - using ``tmp`` got lines to fit.

I would probably just use ``output`` or ``_out``. I do not think anyone really cares about ``lslpp``.

Reply:
Well, I made comments based on the maintainability assumptions. I don't know if anybody cares, though

Reply:
The other issue I kept running into - over various versions - is that ``subprocess`` has seen many improvements over the different versions, and iirc ``test=True`` is not always available.

I'll look into it though - applying ``.decode()`` was to get the string back to what I expected versus what was there. I find it easier to read without ``.decode()``

Reply:
Switched to `out` as return variable. It is just output with several fields separated by `':''`. No more exciting than other platforms implementation data. If needed for maintainability I could add some comments to describe the output.

Reply:
Also, did not add 'text=True' as this failed on Python3-3.5 and still trying to have one source baseline for Cpython and pypa/pip and pypa/packaging

Reply:
The old spelling was "universal_newlines=True" (we added "text=True" to improve both clarity and brevity).

The explicit decode() call is fine, though.